snakemake/workflow.py

```diff
@@ -670,6 +663,13 @@ def files(items):
             )
             return False
 
+        logger.info("Building DAG of jobs...")
+        dag.init()
+        dag.update_checkpoint_dependencies()
+        # check incomplete has to run BEFORE any call to postprocess
+        dag.check_incomplete()
+        dag.check_dynamic()
+
```

Comment:
This has been moved to the wrong place. `self.persistence.lock()` above needs the DAG to be fully computed.

Reply:
Thanks for pointing this out. Corrected.